src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +11,26 @@
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        Lock writeLock = lock.writeLock();
+        writeLock.lock();
+        try {
+            list.add(element);
+        } finally {
+            writeLock.unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
-        return null;
+        Lock readLock = lock.readLock();
+        readLock.lock();
+        try {
+            return list.get(index);
+        } finally {
+            readLock.unlock();
+        }
     }
 
     public int size() {
-        // write your code here
-        return 0;
+        return list.size();
```

Comment:
is this a read operation as well? we are getting some info about the state of the list